### PR TITLE
chg: ignore state-sync tx/receipt while calculating tx/receipt proof

### PR DESCRIPTION
**Pull request by jdkanani**

### CODE REVIEW

File: packages/plasma/src/proofs.ts [hunk 1]
```diff
@@ -176,3 +191,20 @@ export function getReceiptBytes(receipt: TransactionReceipt): Buffer {
...
+// but it is not included in transactionRoot or receiptRoot. 
+// So, while calculating proof, we have to exclude them.
+//
+// This is derived from block's hash and number
+// state-sync tx hash = keccak256("matic-bor-receipt-" + block.number + block.hash)
+export function getStateSyncTxHash(block: Block): Buffer {
+  return keccak256(
+    Buffer.concat([
+      toBuffer("matic-bor-receipt-"), // prefix for bor receipt
+      setLengthLeft(toBuffer(block.number), 8), // 8 bytes of block number (BigEndian)
```
Comment (itzmeanjan) [inline packages/plasma/src/proofs.ts:206]: Is this simply `block.number.toString(16).padStart(64, '0')` ?

**Review by itzmeanjan — APPROVED** [review]
Looks fine, just a question in comment.

File: packages/plasma/src/proofs.ts (hunk 1 above)
Comment (jdkanani) [reply packages/plasma/src/proofs.ts:206]: Kind of -- `block.number.toString(16).padStart(16, '0')` but gives in buffer bytes. Not string.

Comment (itzmeanjan) [reply packages/plasma/src/proofs.ts:206]: Ok got it.

**Review by imyourm8 — APPROVED** [review]

**Review by itzmeanjan — APPROVED** [review]

**Pull request merged**